pkg/parser/whitelist.go

```diff
@@ -19,91 +19,103 @@ type Whitelist struct {
+		return isWhitelisted, false
+	}
+	for _, src := range srcs {
+		if isWhitelisted {
+			break
+		}
+		for _, v := range n.Whitelist.B_Ips {
+			if v.Equal(src) {
+				n.Logger.Debugf("Event from [%s] is whitelisted by IP (%s), reason [%s]", src, v, n.Whitelist.Reason)
+				isWhitelisted = true
 				break
 			}
-			for _, v := range W.B_Ips {
-				if v.Equal(src) {
-					W.Node.Logger.Debugf("Event from [%s] is whitelisted by IP (%s), reason [%s]", src, v, W.Reason)
-					isWhitelisted = true
-					break
-				}
-				W.Node.Logger.Tracef("whitelist: %s is not eq [%s]", src, v)
-			}
-			for _, v := range W.B_Cidrs {
-				if v.Contains(src) {
-					W.Node.Logger.Debugf("Event from [%s] is whitelisted by CIDR (%s), reason [%s]", src, v, W.Reason)
-					isWhitelisted = true
-					break
-				}
-				W.Node.Logger.Tracef("whitelist: %s not in [%s]", src, v)
+			n.Logger.Tracef("whitelist: %s is not eq [%s]", src, v)
+		}
+		for _, v := range n.Whitelist.B_Cidrs {
+			if v.Contains(src) {
+				n.Logger.Debugf("Event from [%s] is whitelisted by CIDR (%s), reason [%s]", src, v, n.Whitelist.Reason)
+				isWhitelisted = true
+				break
 			}
+			n.Logger.Tracef("whitelist: %s not in [%s]", src, v)
 		}
-		hasWhitelist = true
+	}
+	return isWhitelisted, true
+}
+
+func (n *Node) CheckExprWL(cachedExprEnv map[string]interface{}) (bool, bool, error) {
+	err := error(nil)
+	output := interface{}(nil)
+
+	isWhitelisted := false
+
+	if !n.ContainsExprLists() {
+		return false, false, nil
 	}
 	/* run whitelist expression tests anyway */
-	for eidx, e := range W.B_Exprs {
+	for eidx, e := range n.Whitelist.B_Exprs {
 		//if we already know the event is whitelisted, skip the rest of the expressions
 		if isWhitelisted {
 			break
 		}
 		output, err = expr.Run(e.Filter, cachedExprEnv)
 		if err != nil {
-			W.Node.Logger.Warningf("failed to run whitelist expr : %v", err)
-			W.Node.Logger.Debug("Event leaving node : ko")
+			n.Logger.Warningf("failed to run whitelist expr : %v", err)
+			n.Logger.Debug("Event leaving node : ko")
 			break
 		}
 		switch out := output.(type) {
 		case bool:
-			if W.Node.Debug {
-				e.ExprDebugger.Run(W.Node.Logger, out, cachedExprEnv)
+			if n.Debug {
+				e.ExprDebugger.Run(n.Logger, out, cachedExprEnv)
 			}
 			if out {
-				W.Node.Logger.Debugf("Event is whitelisted by expr, reason [%s]", W.Reason)
+				n.Logger.Debugf("Event is whitelisted by expr, reason [%s]", n.Whitelist.Reason)
 				isWhitelisted = true
 			}
-			hasWhitelist = true
 		default:
-			W.Node.Logger.Errorf("unexpected type %t (%v) while running '%s'", output, output, W.Exprs[eidx])
+			n.Logger.Errorf("unexpected type %t (%v) while running '%s'", output, output, n.Whitelist.Exprs[eidx])
 		}
 	}
-	return isWhitelisted, hasWhitelist, err
+	return isWhitelisted, true, err
 }
 
-func (W *Whitelist) Compile(n *Node) (bool, error) {
-	for _, v := range W.Ips {
-		W.B_Ips = append(W.B_Ips, net.ParseIP(v))
+func (n *Node) Compile() (bool, error) {
```

Comment:
Can we call it CompileWhitelist ? So it more defined what it compiles as node has many properties?